File: dotnet_api/src/Modules/AdminDashboard/DTOs/AdminDashboardDTOs.cs
using WebApp_API.Modules.Orders.DTOs;
using WebApp_API.Modules.OrderItems.DTOs;

namespace WebApp_API.Modules.AdminDashboard.DTOs
{
    public class LineChartPoint
    {
        public DateTime Date { get; set; }
        public int OrderCount { get; set; }
        public decimal TotalRevenue { get; set; }
    }

    public class AdminDashboardDTOs
    {
        public int TotalUsers { get; set; }
        public int TotalOrders { get; set; }
        public decimal TotalRevenue { get; set; }
        public List<RecentOrderDto>? RecentOrders { get; set; }
        public List<TopProductDto>? TopSellingProducts { get; set; }
        public List<Entities.Product>? TopNewestProducts { get; set; }
        public List<LineChartPoint>? LineChartPoints { get; set; }
    }
}

File: react-frontend/src/types/dto/AdminDashboardDTOs.ts
import type { Product } from "../models/products/Product";

export interface RecentOrder {
  id: number;
  customerName: string;
  totalAmount: number;
  status: string;
}

export interface TopSellingProduct {
  productId: number;
  productName: string;
  unitPrice: number;
  totalQuantity: number;
  totalRevenue: number;
}

export interface LineChartPoints {
  date: Date;
  orderCount: number;
  totalRevenue: number;
}

export interface DashboardSummary {
  totalUsers: number;
  totalOrders: number;
  totalRevenue: number;
  recentOrders: RecentOrder[];
  topsellingProducts: TopSellingProduct[];
  topNewestProducts: Product[];
  lineChartPoints: LineChartPoints[];
}